Client/Frontend/Browser/Tab Tray/TabTrayController+SearchResultDelegate.swift

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+
+// MARK: UISearchResultUpdating
+
+extension TabTrayController: UISearchResultsUpdating {
+
+  func updateSearchResults(for searchController: UISearchController) {
+    guard let query = searchController.searchBar.text else { return }
+    
+    invalidateSearchTimer()
+    
+    searchTabTrayTimer = Timer.scheduledTimer(
+      timeInterval: 0.1,
+      target: self,
+      selector: #selector(fetchSearchResults(timer:)),
+      userInfo: query,
+      repeats: false)
+  }
+
+  @objc private func fetchSearchResults(timer: Timer) {
+    guard let query = timer.userInfo as? String else {
+      tabTraySearchQuery = ""
```

Comment:
nit: I think this property can be an optional string and default to nil instead of empty string?

Reply:
Great call. Fixed at https://github.com/brave/brave-ios/pull/5809/commits/8f2607664aadd1db8ecadc85dd584cb748433a79